nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -131,15 +133,25 @@ public static void checkReceivedMail(List<MailMessage> mails, DocumentModel comm
 
     public static String getExpectedMailContent(DocumentModel commentDocModel, DocumentModel commentedDocModel,
             Event event, String commentEventType) {
-        URL url = CommentUtils.class.getResource("/templates/commentNotificationMail.txt");
         try {
-            String content = Files.readString(Paths.get(url.toURI()));
-            var model = Map.of("COMMENT_AUTHOR", commentDocModel.getPropertyValue(COMMENT_AUTHOR), //
-                    "COMMENT_ACTION", COMMENT_ADDED.equals(commentEventType) ? "added" : "updated", //
-                    "COMMENTED_DOCUMENT", commentedDocModel.getName(), //
-                    "COMMENT_DATE", EVENT_DATE_FORMAT.format(Date.from(Instant.ofEpochMilli(event.getTime()))), //
-                    "COMMENT_TEXT", commentDocModel.getPropertyValue(COMMENT_TEXT), //
-                    "COMMENT_SUBSCRIPTION_NAME", COMMENT_ADDED.equals(commentEventType) ? "New" : "Updated");
+            var model = new HashMap<String, Serializable>();
+            model.put("COMMENT_AUTHOR", commentDocModel.getPropertyValue(COMMENT_AUTHOR));
+            model.put("COMMENT_ACTION", COMMENT_ADDED.equals(commentEventType) ? "added" : "updated");
+            model.put("COMMENTED_DOCUMENT", commentedDocModel.getName());
+            model.put("COMMENT_DATE", EVENT_DATE_FORMAT.format(Date.from(Instant.ofEpochMilli(event.getTime()))));
+            model.put("COMMENT_TEXT", commentDocModel.getPropertyValue(COMMENT_TEXT));
+            model.put("COMMENT_SUBSCRIPTION_NAME", COMMENT_ADDED.equals(commentEventType) ? "New" : "Updated");
+
+            String template = "commentNotificationMail.txt";
+            DocumentModel parentComment = (DocumentModel) event.getContext().getProperties().get(PARENT_COMMENT);
+            if (COMMENT_DOC_TYPE.equals(parentComment.getType()) ||
+                    ANNOTATION_DOC_TYPE.equals(parentComment.getType())) {
```

Comment:
you can use instead it the `hasSchema` if you want as both annotation and comment contains the comment schema 

for more further  i was think if we put check only here i mean in this method and just rely on the existence of the value in the `ftl` to avoid making the same thing twice see https://github.com/nuxeo/nuxeo/pull/3664/files#diff-b4cc6f294fb9a7dcb604d5682fb14bcaR27 why if we rely on the existence of `Map#PARENT_COMMENT_AUTHOR` and `Map#PARENT_COMMENT_TEXT` to add the section in the `ftl`

Reply:
Map#PARENT_COMMENT_TEXT is not reachable from the ftl
the ftl always have the parentComment which might be the top level doc. So I need to check its type there to add the extra text or not